widgets/common/src/hooks/useEventListener.js

```diff
@@ -0,0 +1,14 @@
+import { useEffect } from 'react';
+
+function useEventListener(toolbox, event, handler = toolbox.refresh) {
```

Comment:
How about using TypeScript for this new hook?

Reply:
I'm not sure `toolbox.refresh` is the best default value here. I don't think there is a good default value, actually :smile: I agree it's the most **common** one, but when I saw `useEventListener`, I would not immediately know that it would cause a refresh when the event occurred.

I encourage making the `handler` required

```suggestion
function useEventListener(toolbox, event, handler) {
```

Reply:
> How about using TypeScript for this new hook?

I'm not sure how should we handle type for `toolbox`, I can see in `renderDeploymentRow.tsx` you are importing it directly from `StageApi`, but is this the way to go? I guess `StageApi` should be part of `Stage` to adhere to how we share things across `app` and `widgets`. Anyway, for me it seems this area is not settled yet, so I'd rather not mess with it at the moment.

Reply:
> I'm not sure `toolbox.refresh` is the best default value here.

Actually, after revisiting how this hook is used it makes perfect sense to name it `useRefreshEvent`, what I just did. With such name it's no longer surprising that `toolbox.refresh` is the default handler.

Reply:
> Anyway, for me it seems this area is not settled yet, so I'd rather not mess with it at the moment.

Sure, that's fine :slightly_smiling_face: 

---

About `useRefreshEvent` - good idea :smile: the default value makes sense now :slightly_smiling_face: